jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -720,25 +718,21 @@ public FileVisitResult visitFile(Path f, BasicFileAttributes attrs) throws IOExc
      */
     public Collection<URL> getTlds(URI uri) throws IOException
     {
-        HashSet<URL> tlds = new HashSet<URL>();
-
+        HashSet<URL> tlds = new HashSet<>();
         URI jarUri = uriJarPrefix(uri, "!/");
-        URL url = jarUri.toURL();
-        JarURLConnection jarConn = (JarURLConnection)url.openConnection();
-        jarConn.setUseCaches(Resource.getDefaultUseCaches());
-        JarFile jarFile = jarConn.getJarFile();
-        Enumeration<JarEntry> entries = jarFile.entries();
-        while (entries.hasMoreElements())
-        {
-            JarEntry e = entries.nextElement();
-            String name = e.getName();
-            if (name.startsWith("META-INF") && name.endsWith(".tld"))
+        try (Resource.Mount mount = Resource.mount(jarUri);
+             Stream<Path> stream = Files.walk(mount.root().getPath()))
+        {
+            Iterator<String> it = stream
+                .map(Path::toString)
+                .filter(filename -> filename.startsWith("META-INF") && filename.endsWith(".tld"))
```

Comment:
Create a new static method in this class ...

``` java
    public static boolean isTldFile(Path path)
    {
        if (path.getNameCount() < 2)
            return false;
        if (!path.getName(0).toString().equalsIgnoreCase("META-INF"))
            return false;
        String filename = path.getFileName().toString();
        return filename.toLowerCase(Locale.ENGLISH).endsWith(".tld");
    }
```

And use it in place of this filter.